teamscale_precommit_client/precommit_client.py

```diff
@@ -1,37 +1,41 @@
 from __future__ import absolute_import
-from __future__ import unicode_literals
 from __future__ import print_function
+from __future__ import unicode_literals
 
+import argparse
 import datetime
-import time
 import os
 import sys
-import argparse
+import time
 
-from teamscale_precommit_client.git_utils import get_current_branch, get_current_timestamp
-from teamscale_precommit_client.git_utils import get_changed_files_and_content, get_deleted_files
-from teamscale_precommit_client.data import PreCommitUploadData
 from teamscale_client import TeamscaleClient
+
 from teamscale_precommit_client.client_configuration_utils import get_teamscale_client_configuration
+from teamscale_precommit_client.data import PreCommitUploadData
+from teamscale_precommit_client.git_utils import get_changed_files_and_content, get_deleted_files
+from teamscale_precommit_client.git_utils import get_current_branch, get_current_timestamp
 from teamscale_precommit_client.git_utils import get_repo_root_from_file_in_repo
 
 # Filename of the precommit configuration. The client expects this config file at the root of the repository.
 PRECOMMIT_CONFIG_FILENAME = '.teamscale-precommit.config'
+DEFAULT_PROJECT_SUBPATH = ''
 
 
 class PrecommitClient:
     """Client for precommit analysis"""
     # Number of seconds the client waits until fetching precommit results from the server.
     PRECOMMIT_WAITING_TIME_IN_SECONDS = 2
 
-    def __init__(self, teamscale_config, repository_path, analyzed_file=None, verify=True,
-                 omit_links_to_findings=False, exclude_findings_in_changed_code=False, fetch_existing_findings=False,
-                 fetch_all_findings=False, fetch_existing_findings_in_changes=False, fail_on_red_findings=False,
-                 log_to_stderr=False):
+    def __init__(self, teamscale_config, repository_path, project_subpath='', analyzed_file=None,
+                 verify=True, omit_links_to_findings=False, exclude_findings_in_changed_code=False,
+                 fetch_existing_findings=False, fetch_all_findings=False, fetch_existing_findings_in_changes=False,
+                 fail_on_red_findings=False, log_to_stderr=False):
         """Constructor"""
         self.teamscale_client = TeamscaleClient(teamscale_config.url, teamscale_config.username,
                                                 teamscale_config.access_token, teamscale_config.project_id, verify)
         self.repository_path = repository_path
+        # os.path.join is required to add a tailing / if it's not already there
```

Comment:
The comment on the prefix branch is different, maybe sync them to make it easier to grasp that they are saying the same?

Reply:
I'd remove on of them in the merge commit so it'd look like:
```
...
self.repository_path = repository_path

# calling os.path.join ensures a tailing '/'
self.path_prefix = os.path.join(path_prefix, '')
self.project_subpath = os.path.join(project_subpath, '')

self.analyzed_file = analyzed_file
...
```

Reply:
Works for me.